File: app/src/test/java/com/mgaetan89/showsrage/fragment/AddShowOptionsFragment_GetLocationTest.java
package com.mgaetan89.showsrage.fragment;

import android.widget.Spinner;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@RunWith(Parameterized.class)
public class AddShowOptionsFragment_GetLocationTest {
	@Parameterized.Parameter(1)
	public String location;

	@Parameterized.Parameter(0)
	public Spinner spinner;

	@Test
	public void getLocation() {
		assertThat(AddShowOptionsFragment.Companion.getLocation(this.spinner)).isEqualTo(this.location);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{null, null},
				{getMockedSpinner(null), null},
				{getMockedSpinner(""), ""},
				{getMockedSpinner("/home/videos/Shows"), "/home/videos/Shows"},
		});
	}

	private static Spinner getMockedSpinner(String selectedItem) {
		Spinner spinner = mock(Spinner.class);
		when(spinner.getSelectedItem()).thenReturn(selectedItem);

		return spinner;
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/AddShowOptionsFragment_GetLocationTest.kt
package com.mgaetan89.showsrage.fragment

import android.widget.Spinner
import com.mgaetan89.showsrage.model.RootDir
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

@RunWith(Parameterized::class)
class AddShowOptionsFragment_GetLocationTest(val spinner: Spinner?, val location: String?) {
	@Test
	fun getLocation() {
		assertThat(AddShowOptionsFragment.getLocation(this.spinner)).isEqualTo(this.location)
	}

	companion object {
		@JvmStatic
		@Parameterized.Parameters
		fun data(): Collection<Array<Any?>> {
			return listOf(
					arrayOf<Any?>(null, null),
					arrayOf<Any?>(getMockedSpinner(null), null),
					arrayOf<Any?>(getMockedSpinner(""), ""),
					arrayOf<Any?>(getMockedSpinner("/home/videos/Shows"), "/home/videos/Shows")
			)
		}

		private fun getMockedSpinner(location: String?): Spinner {
			var rootDir: RootDir? = null

			if (location != null) {
				rootDir = RootDir()
				rootDir.location = location
			}

			val spinner = mock(Spinner::class.java)
			`when`(spinner.selectedItem).thenReturn(rootDir)

			return spinner
		}
	}
}
